### DISCUSSION TITLE
Language Parity - Roadmap for C# Support

### DISCUSSION THREAD

**Discussion by SamuelAsherRivello** [MikeSchulze/gdUnit4#520] [General]
I'm a game dev and unit testing expert. 

https://www.samuelasherrivello.com/

I'm new to Godot and have questions about your great plugin.

I see on https://mikeschulze.github.io/gdUnit4/ that some features lack language parity.

# Summary

Areas with GDScript support but not C# support are...

### Testing
* (None that I've noticed)
### Advanced Testing
* https://mikeschulze.github.io/gdUnit4/advanced_testing/fuzzing/#testing-with-fuzzers
* https://mikeschulze.github.io/gdUnit4/advanced_testing/signals/#signal-monitor
* https://mikeschulze.github.io/gdUnit4/advanced_testing/mock/#mocking--mocks
* https://mikeschulze.github.io/gdUnit4/advanced_testing/spy/#spy
* https://mikeschulze.github.io/gdUnit4/advanced_testing/argument_matchers/#argument-matchers

# Questions

1. Is the above summary correct? Anything to add/remove?
2. What is the roadmap/timing on the issues above?

**Comment by MikeSchulze** [comment]
Hi, I don't want to implement stuff that already exists as third party library

> + https://mikeschulze.github.io/gdUnit4/advanced_testing/fuzzing/#testing-with-fuzzers

 you can use e.g. https://github.com/Metalnem/sharpfuzz

> + https://mikeschulze.github.io/gdUnit4/advanced_testing/signals/#signal-monitor

  the AssertSignals already have integrated the monitor see https://github.com/MikeSchulze/gdUnit4Net/issues/121

> + https://mikeschulze.github.io/gdUnit4/advanced_testing/mock/#mocking--mocks

 you can use e.g. https://github.com/devlooped/moq

> + https://mikeschulze.github.io/gdUnit4/advanced_testing/spy/#spy

 you can use e.g. https://github.com/devlooped/moq/wiki/Quickstart#customizing-mock-behavior

> + https://mikeschulze.github.io/gdUnit4/advanced_testing/argument_matchers/#argument-matchers

it is actual not planned to implement

As long as the demand for GdUnit4Net does not increase, it is not a priority to install new things. 
Compared to GdUnit4, you can see from the stars that this is not a priority.
As long as the demand for GdUnit4Net does not increase, it is not a priority to install new things. 
Compared to GdUnit4, you can see from the stars that this is not a priority.


The current work is to enable Rider to debug tests without using a workaround.
https://github.com/MikeSchulze/gdUnit4Net/pull/130
